backend/src/main/java/com/karankumar/bookproject/backend/service/UserService.java

```diff
@@ -137,6 +147,14 @@ public boolean emailIsNotInUse(String email) {
     }
 
     public void changeUserPassword(@NonNull User user, @NonNull String password) {
+        Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
+        Set<ConstraintViolation<String>> constraintViolations = validator.validate(password);
+
+        Strength passwordStrength = zxcvbn.measure(password);
+        if (passwordStrength.getScore() <= 2){
+            throw new ConstraintViolationException("weak password!!", constraintViolations);
```

Comment:
Instead of throwing a ConstraintViolationException, which will result in a 500 internal server error, we should throw a client-side error (in the 400 region) to indicate to the client that the fault is on their end, not in the backend